rcl/src/rcl/timer.c

```diff
@@ -39,7 +39,7 @@ struct rcl_timer_impl_s
   // The user supplied callback.
   atomic_uintptr_t callback;
   // This is a duration in nanoseconds.
-  atomic_uint_least64_t period;
+  atomic_int_least64_t period;
```

Comment:
this is initialized with `int64_t`, and used for calculation internally.

https://github.com/ros2/rcl/blob/3fa178ad6c60bf16447654a01a4f68e8d400b517/rcl/src/rcl/timer.c#L127-L131

Reply:
Can you please add a comment here explaining that?  That way we won't have to have this conversation again :).

Reply:
@clalancette added comment with https://github.com/ros2/rcl/pull/1010/commits/c295170d557f8b32354d15fc2bae30eb6110709e